java/lance-namespace-hms/src/main/java/com/lancedb/lance/namespace/ClientPool.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.lancedb.lance.namespace;
+
+// Copied from apache iceberg.
```

Comment:
would be good to add a reference link for the actual code, and also update the NOTICE like this: https://github.com/apache/iceberg/blob/main/LICENSE#L216-L223